tests/foreman/api/test_subnet.py

```diff
@@ -0,0 +1,399 @@
+
+        :id: 64ac7873-ed36-4cad-a8db-5c9a4826868b
+
+        :steps:
+
+            1. Create valid subnet with valid parameter
+            2. Create host with the above subnet
+            3. Update subnet parameter with new name and value
+
+        :expectedresults:
+
+            1. The inherited subnet parameter in host should have
+                updated name and value
+            2. The inherited subnet parameter in host enc should have
+                updated name and value
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_delete_subnet_parameter(self):
+        """Subnet parameter can be deleted
+
+        :id: 972b66ec-d506-4fcb-9786-c62f2f79ac1a
+
+        :steps:
+
+            1. Create valid subnet with valid parameter
+            2. Delete the above subnet parameter
+
+        :expectedresults: The parameter should be deleted from subnet
+        """
+
+    @stubbed
+    @skip_if_bug_open('bugzilla', 1470014)
+    @tier2
+    def test_positive_delete_subnet_parameter_host_impact(self):
+        """Deleting parameter from subnet component deletes the parameter in
+        host inheriting that subnet
+
+        :id: f5174e00-33cd-4008-8112-9c8190859244
+
+        :steps:
+
+            1. Create valid subnet with valid parameter
+            2. Create host with the above subnet
+            3. Delete the above parameter from subnet
+            4. List subnet parameters for above host
+
+        :expectedresults:
+
+            1. The parameter should be deleted from host
+            2. The parameter should be deleted from host enc
+        """
+
+    @stubbed
+    @skip_if_bug_open('bugzilla', 1470014)
+    @tier2
+    def test_positive_delete_subnet_overrided__parameter_host_impact(self):
+        """Deleting parameter from subnet component doesnt deletes its
+        overridden parameter in host inheriting that subnet
+
+        :id: e8f4a8e8-64ec-4a0f-aee4-14b6e984b470
+
+        :steps:
+
+            1. Create valid subnet with valid parameter
+            2. Create host with the above subnet
+            3. Override subnet parameter value from host
+            4. Delete the above parameter from subnet
+            5. List host parameters
+
+        :expectedresults:
+
+            1. The parameter should not be deleted from host as it becomes
+                host parameter now
+            2. The parameter should not be deleted from host enc as well
+        """
+
+    @stubbed
+    @tier2
+    def test_positive_list_parameters(self):
+        """Satellite lists all the subnet parameters
+
+        :id: ce86d531-bf6b-45a9-81e3-67e1b3398f76
+
+        :steps:
+
+            1. Create subnet with all the details
+            2. Add two parameters in subnet
+            3. List parameters of subnet
+
+        :expectedresults: The satellite should display all the subnet
+            parameters
+        """
+
+    @stubbed
+    @tier2
+    def test_positive_info_parameter(self):
```

Comment:
Redundant test case as you will `.read()` subnet parameters in all other test cases to be able to verify expected results